clients/admin-ui/src/features/config-wizard/ScanResultsForm.tsx

```diff
@@ -51,7 +62,17 @@ const ScanResultsForm = () => {
   );
 
   const handleSubmit = (values: FormValues) => {
-    dispatch(chooseSystemsForReview(values.selectedKeys));
+    if (systems.length > values.selectedKeys.length) {
+      setSelectedKeyValues(values.selectedKeys);
+      onOpen();
+    } else {
+      dispatch(chooseSystemsForReview(values.selectedKeys));
+      dispatch(changeStep());
+    }
+  };
+
+  const confirmRegisterSelectedSystems = () => {
+    dispatch(chooseSystemsForReview(selectedKeyValues));
```

Comment:
Does it make sense to use the batch() hook here given a single render pass?

Reply:
I think we're still on react 17?

Reply:
Yes React 17.0.2